src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -261,10 +267,59 @@ public static void parseEditStep(Integer recipeIndex, String description) throws
             throw new Exception("error in edit step:\n" + e.getMessage());
         }
     }
-    public static String removeForbiddenChars(String ingredient) {
-        for (String chara : StringLib.FORBIDDEN_CHARS) {
-            ingredient.replaceAll(chara, "");
+    public static String[] parseAddToRecipeDescription(String description) {
+        String[] subDescriptions = description.split("/", 3);
+        String element0 = subDescriptions[0].toLowerCase().split("id")[0].trim();
+        subDescriptions[0] = element0.replace("-","");
+        String element1 = subDescriptions[1].toLowerCase().split("desc")[0].trim();
+        subDescriptions[1] = element1;
+        return subDescriptions;
+    }
+    public static String[] parseDeleteFromRecipeDescription(String description) {
+        String[] subDescriptions = description.split("/", 2);
+        String element0 = subDescriptions[0].toLowerCase().split("id")[0].trim();
+        subDescriptions[0] = element0.replace("-","");
+        return subDescriptions;
+    }
+    public static boolean isValidAddToRecipe(String description) {
+        String descLowerCase = description.toLowerCase().trim();
+        if ((descLowerCase.contains("step") ||
+                descLowerCase.contains("ingredient")) &&
+                descLowerCase.contains("id/") &&
+                descLowerCase.contains("desc/") &&
+                !(descLowerCase.contains("step") &&
+                        descLowerCase.contains("ingredient"))) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+    public static boolean isValidDeleteFromRecipe(String description) {
+        String descLowerCase = description.toLowerCase().trim();
+        if ((descLowerCase.contains("step") ||
```

Comment:
you should check only one step input. and the order is correct. this checking method is not enough.